### DISCUSSION TITLE
Cant load file with non ascii characters

### DISCUSSION THREAD

**Discussion by Tem4a** [community/community#46159] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Cant load file through github api to the repository (curl.php) with non ascii characters in name (example : 米津玄師 , — , and other non english letter).

use curl like this :

	

	$data_git = array( "message"=>"Required message", "content"=> base64_encode( "content" ));

	$ch_git = curl_init('https://api.github.com/repos/OWNER/REPO/contents/PATH/'.$name);
	curl_setopt($ch_git, CURLOPT_CUSTOMREQUEST, "PUT");
	curl_setopt($ch_git, CURLOPT_POSTFIELDS, $data_string_git);
	curl_setopt($ch_git, CURLOPT_RETURNTRANSFER, true);
	
	curl_setopt($ch_git, CURLOPT_HTTPHEADER, array(
		"Accept: application/vnd.github+json" , 
		"Authorization: Bearer ".$github_key ,
		"X-GitHub-Api-Version: 2022-11-28" ,
		'User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 YaBrowser/19.9.3.314 Yowser/2.5 Safari/537.36',
	));

	$result_git = curl_exec($ch_git);

What could be the problem ? Maybe some headers is needed?

**Comment by Ben-Mack** [comment]
I'm having the same problem too with using curl.exe directly, not just filename, but it will failed with error `curl: (26) Failed to open/read local data from file/application` if non-ASCII character present in the path/folder name.
Hope to have some solution

**Comment by jge162** [comment]
Hello and good day @Frederik123 @Ben-Mack 

You can try encoding the file path using the `cmd /c echo` command with the `chcp` command to set the code page to UTF-8:

cmd /c echo "file path with non-ASCII characters" | curl -K -

chcp 65001

This should properly encode the file path and allow it to be processed correctly by curl.exe

**Comment by Tem4a** [comment, marked as answer]
Well , i make some research and find solution. 

https://github.com/KnpLabs/php-github-api/blob/master/lib/Github/Api/Repository/Contents.php#L103
Need to encode file name with "rawurlencode" (https://www.php.net/manual/en/function.rawurlencode.php) . 
Then file upload without problems.
